idaes/apps/grid_integration/tests/test_PEM_Parameterized_bidder.py

```diff
@@ -0,0 +1,198 @@
+    pem_pmax = bidder_object.pem_mw
+    pem_marginal_cost = bidder_object.pem_marginal_cost
+    date = "2020-01-01"
+
+    # test DA bidding
+    bids = bidder_object.compute_day_ahead_bids(date=date, hour=0)
+    expected_DA_cf = [0.1, 0.1, 0.1, 0.8, 0.8, 0.8]
+    expected_bids = {}
+    for t in range(day_ahead_horizon):
+        expect_da_wind = expected_DA_cf[t] * pmax
+        if t <= 2:
+            expect_bids_curve = [(0, 0), (expected_DA_cf[t] * pmax, pem_marginal_cost)]
+        else:
+            expect_bids_curve = [
+                (0, 0),
+                (expected_DA_cf[t] * pmax - pem_pmax, 0),
+                (expected_DA_cf[t] * pmax, pem_marginal_cost),
+            ]
+        expect_cost_curve = convert_marginal_costs_to_actual_costs(expect_bids_curve)
+        expected_bids[t] = {
+            gen: {
+                "p_cost": expect_cost_curve,
+                "p_min": pmin,
+                "p_max": expect_da_wind,
+                "startup_capacity": expect_da_wind,
+                "shutdown_capacity": expect_da_wind,
+            }
+        }
+    pyo_unittest.assertStructuredAlmostEqual(first=expected_bids, second=bids)
+
+
+@pytest.mark.component
+def test_compute_RT_bids(bidder_object):
+    """
+    This is to test the if the compute_DA_bids function works correctly.
+    """
+    gen = bidder_object.generator
+    pmin = bidder_object.bidding_model_object.pmin
+    pmax = bidder_object.bidding_model_object.pmax
+    pem_pmax = bidder_object.pem_mw
+    pem_marginal_cost = bidder_object.pem_marginal_cost
+    date = "2020-01-01"
+    # totally 3 cases:
+    # rt_wind - realized_day_ahead_dispatches > P_pem_max (t = 0)
+    # 0 <= rt_wind - realized_day_ahead_dispatches <= P_pem_max (t = 1, 2, 3, 4)
+    # rt_wind <= realized_day_ahead_dispatches (t = 5)
+    realized_day_ahead_dispatches = [20, 20, 20, 120, 120, 180]
+    realized_day_ahead_prices = None
+
+    # testing RT bidding
+    bids = bidder_object.compute_real_time_bids(
+        date=date,
+        hour=0,
+        realized_day_ahead_dispatches=realized_day_ahead_dispatches,
+        realized_day_ahead_prices=realized_day_ahead_prices,
+    )
+
+    expected_RT_cf = [1.0, 0.2, 0.2, 0.75, 0.75, 0.75]
+    expected_bids = {}
+    for t in range(real_time_horizon):
+        expect_rt_wind = expected_RT_cf[t] * pmax
+        if t == 0:
+            expect_bids_curve = [
+                (0, 0),
+                (expect_rt_wind - realized_day_ahead_dispatches[t] - pem_pmax, 0),
+                (expect_rt_wind - realized_day_ahead_dispatches[t], pem_marginal_cost),
+            ]
+        elif t in [1, 2, 3, 4]:
+            expect_bids_curve = [
+                (0, 0),
+                (expect_rt_wind - realized_day_ahead_dispatches[t], pem_marginal_cost),
+            ]
+        else:
+            expect_bids_curve = [(0, 0)]
+        expect_cost_curve = convert_marginal_costs_to_actual_costs(expect_bids_curve)
+        expected_bids[t] = {

```

Comment:
Same comment as above

Reply:
Hi Darice, thank you for spending time reviewing the codes! I have addressed all your comments today.